### PR TITLE
Fix json editor

### PR DESCRIPTION
**Pull request by Ujjwal-UJ**
Closes #
- Closes https://github.ibm.com/wiotp/Maximo-Asset-Monitor/issues/2711

**Summary**

- Fixed Open JSON editor button should always be visible in config pane

**Change List (commits, features, bugs, etc)**

- fix(cardeditor): fixed Open JSON editor button in config pane

**Acceptance Test (how to verify the PR)**

- Open JSON Editor button should be fixed in config pane

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/CardEditor/CardEditForm/CardEditForm.jsx [hunk 1]
```diff
@@ -143,209 +126,59 @@ export const getCardSizeText = (size, i18n) => {
...
-  let attributes;
-  let series;
-  let columns;
-  if (card.content?.attributes) {
-    attributes = card.content.attributes.map((attribute) =>
-      omit(attribute, ['aggregationMethods', 'grain', 'downSampleMethods'])
-    );
-  }
-  if (card.content?.series) {
-    series = card.content.series.map((attribute) =>
```
Comment (davidicus) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditForm.jsx:180]: There seems to be a lot of deletions on this page. Is this on purpose.

File: packages/react/src/components/CardEditor/CardEditForm/CardEditForm.test.jsx [hunk 2]
```diff
@@ -1,55 +1,18 @@
...
-    ],
-  },
-};
 
 afterEach(() => {
   jest.clearAllMocks();
 });
 describe('CardEditForm', () => {
   describe('CardEditForm', () => {
-    it('should fire onCardJsonPreview when opening the json modal', () => {
```
Comment (davidicus) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditForm.test.jsx:38]: Why are we deleting these test? None of these changes have to do with the listed issues, unless I am missing something.

**Review by davidicus — COMMENTED** [review]
There are a lot of changes here that seem unrelated to the fixes mentioned in the description.

File: packages/react/src/components/CardEditor/CardEditForm/CardEditForm.jsx (hunk 1 above)
Comment (Ujjwal-UJ) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditForm.jsx:180]: Yes, these methods are been moved to CardEditor.jsx for bug fix

File: packages/react/src/components/CardEditor/CardEditForm/CardEditForm.test.jsx (hunk 2 above)
Comment (Ujjwal-UJ) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditForm.test.jsx:38]: As the methods are been moved from CardEditForm.jsx to CardEditor.jsx for issue https://github.ibm.com/wiotp/Maximo-Asset-Monitor/issues/2711, we had to move test cases as well to CardEditor.test.jsx

Comment (davidicus) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditForm.test.jsx:38]: This isn't mentioned at all in your description of the PR and there are no issues opened here explaining what is requiring this change.

**Comment by herleraja** [comment]
@davidicus some part of code from packages/react/src/components/CardEditor/CardEditForm/CardEditForm.jsx moved to packages/react/src/components/CardEditor/CardEditor.jsx so do the testcases. Functionality wise nothing much changed. 

Overall use case: The `Open JSON editor` button needed to be in the footer. 
image

File: packages/react/src/components/CardEditor/CardEditor.jsx [hunk 3]
```diff
@@ -248,36 +360,88 @@ const CardEditor = ({
...
+            ) : null}
+            <CardEditForm
+              cardConfig={finalCardToEdit}
+              isSummaryDashboard={isSummaryDashboard}
+              onChange={onChange}
+              dataItems={dataItems}
+              getValidDataItems={getValidDataItems}
+              getValidTimeRanges={getValidTimeRanges}
+              onValidateCardJson={onValidateCardJson}
+              onCardJsonPreview={onCardJsonPreview}
```
Comment (herleraja) [inline packages/react/src/components/CardEditor/CardEditor.jsx:401]: This has been removed from CardEditForm correct ? please remove unwanted props from passing into CardEditForm

**Review by herleraja — APPROVED** [review]
LGTM

File: packages/react/src/components/CardEditor/CardEditor.jsx (hunk 3 above)
Comment (Ujjwal-UJ) [reply packages/react/src/components/CardEditor/CardEditor.jsx]: Updated

File: packages/react/src/components/CardEditor/CardEditor.test.jsx [hunk 4]
```diff
@@ -222,4 +259,191 @@ describe('CardEditor', () => {
       type: 'VALUE',
     });
   });
+
```
Comment (davidicus) [inline packages/react/src/components/CardEditor/CardEditor.test.jsx:262]: FYI we could write cypress test for these now.

File: packages/react/src/components/List/_list.scss [hunk 5]
```diff
@@ -38,6 +38,7 @@
 
         .#{$iot-prefix}-simple-pagination-page-label {
           flex-grow: 2;
+          text-align: right;
```
Comment (davidicus) [inline packages/react/src/components/List/_list.scss:41]: This goes against our design spec for simple pagination and should not be required to fix your defect.

**Review by davidicus — CHANGES_REQUESTED** [review]

File: packages/react/src/components/List/_list.scss (hunk 5 above)
Comment (herleraja) [reply packages/react/src/components/List/_list.scss:41]: @Ujjwal-UJ lets remove this.

Comment (Ujjwal-UJ) [reply packages/react/src/components/List/_list.scss]: removed pagination styles

**Pull request merged**